extensions/blocks/dialogue/dialogue.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Dialogue Block.
+ *
+ * @since 9.3.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Dialogue;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'dialogue';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
+	$uses       = $deprecated ? 'context' : 'uses_context';
+	Blocks::jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			$uses             => array(
+				'jetpack/conversation-participants',
+				'jetpack/conversation-showtimestamp',
+			),
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Dialogue block registration/dependency declaration.
+ *
+ * @param array  $attrs    Array containing the Dialogue block attributes.
+ * @param string $block_content String containing the Dialogue block content.
+ * @param object $block Block object data.
+ *
+ * @return string
+ */
+function render_block( $attrs, $block_content, $block ) {
```

Comment:
I think I would break `render_block` a bit to make it more readable. Ideally, the function would only render the block; the actual data massaging would happen in a separate function.

That may also make things easier to test, which may be useful here since we do a lot of data massaging before the actual display.